app/scripts/components/common/mapbox/map-options/use-basemap.ts

```diff
@@ -0,0 +1,100 @@
+import { Layer, Style } from 'mapbox-gl';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  BasemapId,
+  BASEMAP_STYLES,
+  getStyleUrl,
+  GROUPS_BY_OPTION,
+  Option
+} from './basemaps';
+
+export function useBasemap() {
+  const [basemapStyleId, setBasemapStyleId] = useState<BasemapId>('satellite');
+
+  const onBasemapStyleIdChange = useCallback((basemapId) => {
+    setBasemapStyleId(basemapId);
+  }, []);
+
+  const [baseStyle, setBaseStyle] = useState<Style | undefined>(undefined);
+
+  useEffect(() => {
+    const mapboxId = basemapStyleId
+      ? BASEMAP_STYLES.find((b) => b.id === basemapStyleId)!.mapboxId
+      : BASEMAP_STYLES[0].mapboxId;
+
+    const url = getStyleUrl(mapboxId);
+    const controller = new AbortController();
+
+    const load = async () => {
+      const styleRaw = await fetch(url, { signal: controller.signal });
+      const styleJson = await styleRaw.json();
+      setBaseStyle(styleJson as Style);
+    };
```

Comment:
[Now fetching basemap styles with react-query](https://github.com/NASA-IMPACT/veda-ui/pull/458/commits/777c38e91205a2fc90ec8b2c9feb61e6eb1b1efb)

However, it's unclear to me:
- whether caching actually works. Here style gets reloaded on basemap change, but could it be because of [dev tools](https://react-query-v3.tanstack.com/guides/important-defaults). Maybe something off on my side.
- what's the benefit of react-query when data gets cached on disk by the browser anyways
- whether we should manually cancel fetches when unmounting, since react-query seems to consider that keeping on loading the data is a [sensible default](https://react-query-v3.tanstack.com/guides/query-cancellation)

Thanks 🙏